cmake/modules/FindPySpark.cmake

```diff
@@ -0,0 +1,74 @@
+#
+
+message(STATUS "Looking for pyspark dependency: Java")
+find_package(Java 1.8 COMPONENTS Runtime)
+
+if (Java_Runtime_FOUND)
+    message(STATUS "Found Java ${Java_JAVA_EXECUTABLE}")
+    message(STATUS "Java version ${Java_VERSION_STRING}")
+    message(STATUS "Looking for pyspark dependency: py4j")
+
+    execute_process(
+        COMMAND ${PYTHON_EXECUTABLE_Development_Main} -c "import py4j"
+        RESULT_VARIABLE _PY4J_SEARCH_SUCCESS
+        OUTPUT_VARIABLE _PY4J_VALUES_OUTPUT
+        ERROR_VARIABLE _PY4J_ERROR_VALUE
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    if(${_PY4J_SEARCH_SUCCESS} EQUAL 0)
+        message(STATUS "Package 'py4j' has been succesfully imported with ${PYTHON_EXECUTABLE_Development_Main}")
+        message(STATUS "Import python package: pyspark")
+
+        execute_process(
+            COMMAND ${PYTHON_EXECUTABLE_Development_Main} -c "import pyspark; print(pyspark.__version__)"
+            RESULT_VARIABLE _PYSPARK_SEARCH_SUCCESS
+            OUTPUT_VARIABLE _PYSPARK_VALUES_OUTPUT
+            ERROR_VARIABLE _PYSPARK_ERROR_VALUE
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+
+        string(REGEX MATCH "^[0-9]+\\.[0-9]+\\.[0-9]+" _PYSPARK_VERSION_STRING "${_PYSPARK_VALUES_OUTPUT}")
+
+        if(${_PYSPARK_SEARCH_SUCCESS} EQUAL 0)
+
+            if(${PYTHON_VERSION_STRING_Development_Main} VERSION_LESS "3.6")
+                # Pyspark dropped support for Python < 3.6 in version 3.1.1
+                if(${_PYSPARK_VERSION_STRING} VERSION_GREATER_EQUAL "3.1.1")
+                    message(STATUS "Package 'pyspark' found with version ${_PYSPARK_VERSION_STRING}, which requires Python > 3.5")
+                    message(STATUS "Please install a version of 'pyspark' compatible with your Python executable")
+                    set(PYSPARK_FOUND FALSE)
+                else()
+                    message(STATUS "Package 'pyspark' has been succesfully imported with ${PYTHON_EXECUTABLE_Development_Main}")
+                    set(PYSPARK_FOUND TRUE)
+                endif()
+            else()
+                message(STATUS "Package 'pyspark' has been succesfully imported with ${PYTHON_EXECUTABLE_Development_Main}")
+                set(PYSPARK_FOUND TRUE)
+            endif()
+
+        else()
+            message(STATUS "Error while importing package 'pyspark' with ${PYTHON_EXECUTABLE_Development_Main}\n${_PYSPARK_ERROR_VALUE}")
+            message(STATUS "Please make sure 'pyspark' is installed, e.g. with 'pip install pyspark' or 'conda install -c conda-forge pyspark'")
+            set(PYSPARK_FOUND FALSE)
+        endif()
+    else()
+        message(STATUS "Error while importing package 'py4j' with ${PYTHON_EXECUTABLE_Development_Main}\n${_PY4J_ERROR_VALUE}")
+        message(STATUS "Please make sure 'py4j' is installed, e.g. with 'pip install py4j' or 'conda install -c conda-forge py4j'")
+        set(PYSPARK_FOUND FALSE)
+    endif()
+else()
+    set(PYSPARK_FOUND FALSE)
```

Comment:
Shouldn't `PYSPARK_FOUND` be cached?

Reply:
Following Oksana's suggestion, the [find_package_handle_standard_args](https://cmake.org/cmake/help/v3.9/module/FindPackageHandleStandardArgs.html) function is a better way to signal CMake that pyspark has been found correctly. It will internally set the `PySpark_FOUND` variable